### PR TITLE
fix(@angular-devkit/build-angular): improve builder phase reporting

### PR DESCRIPTION
**Pull request by alan-agius4**
Oct-23-2020 15-20-47

image

### CODE REVIEW

**Review by clydin — APPROVED** [review]

**Review by clydin — APPROVED** [review]

File: packages/angular_devkit/build_angular/src/webpack/configs/common.ts [hunk 1]
```diff
@@ -301,7 +308,23 @@ export function getCommonConfig(wco: WebpackConfigOptions): Configuration {
...
-    extraPlugins.push(new ProgressPlugin({ profile: buildOptions.verbose }));
+    const spinner = new Spinner();
+
+    extraPlugins.push(new ProgressPlugin({
+      handler: (percentage: number, message: string) => {
+        switch (percentage) {
+          case 0:
+            spinner.start(`Generating ${platform} application bundles...`);
+            break;
+          case 1:
```
Comment (kyliau) [inline packages/angular_devkit/build_angular/src/webpack/configs/common.ts:319]: Is the percentage always between 0 and 1?

**Review by kyliau — COMMENTED** [review]
minor comment: Is it possible to add a test to assert all the messages printed by the spinner, for the most common case i.e. the browser build?
If not, that's fine.

File: packages/angular_devkit/build_angular/src/webpack/configs/common.ts (hunk 1 above)
Comment (alan-agius4) [reply packages/angular_devkit/build_angular/src/webpack/configs/common.ts:319]: Yes

File: packages/angular_devkit/build_angular/src/utils/spinner.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  stop(): void {
+    this.spinner.stop();
+  }
+
+  start(text?: string): Spinner {
+    if (this.enabled) {
+      this.spinner.start(text);
+    }
+
+    return this;
```
[the remark this answers is not in the document]
Comment (alan-agius4) [reply packages/angular_devkit/build_angular/src/utils/spinner.ts:45]: Good call, I’ll prefer the later

**Comment by alan-agius4** [comment]
@kyliau, re testing I can add an E2E which can cover most of the logs:

**Review by kyliau — APPROVED** [review]
LGTM.
Re testing: I just wish we could have a way to inspect all the messages we print out at each step in a clear and concise way. If it's too much effort, then we shouldn't bother.

**Comment by filipesilva** [comment]
@alan-agius4 I can't apply this PR cleanly to patch - can you make a new one for patch only please?

**Comment by alan-agius4** [comment]
@filipesilva, here you go https://github.com/angular/angular-cli/pull/19285

**Pull request merged**